flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/schema/FlinkTable.scala

```diff
@@ -44,14 +44,14 @@ abstract class FlinkTable[T](
 
   val fieldTypes: Array[TypeInformation[_]] =
```

Comment:
We can use the util `getFieldTypes` to generate the TypeInformation array to reduce redundant code. And we can put the length check above.